### PR TITLE
Allows code to catch an IndexError if the composer tries to write to …

### PR DESCRIPTION
**Pull request by mitchhung**
…a column that is greater than the index corresponding to Z.

The implementation currently assumes columns are accessed by key (and adds more letters if a `KeyError` is encountered at runtime):

```
     37 def __getitem__(self, item):
     38     try:
---> 39         return self.letters[item]
     40     except KeyError:
     41         self._generate_more_letters(item)
```

However, `self.letters` is actually accessed by `index`, per:
```
     33 @classmethod
     34 def from_location(cls, col, row):
---> 35     col = alphabet[col - 1]
     36     ref = f"{col}{row}"
     37     return cls(ref)
```

### CODE REVIEW

**Review by luke-mino-altherr — APPROVED** [review]

File: src/htmxl/alphabet.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def index(self, value):
     def __getitem__(self, item):
         try:
             return self.letters[item]
-        except KeyError:
+        except (IndexError, KeyError):
```
Comment (luke-mino-altherr) [inline src/htmxl/alphabet.py:40]: testcase to replicate?

**Review by JFCherry — APPROVED** [review]

File: src/htmxl/alphabet.py (hunk 1 above)
Comment (mitchhung) [reply src/htmxl/alphabet.py:40]: added

File: tests/test_compose/test_write/test_alphabet.py [hunk 2]
```diff
@@ -41,3 +41,9 @@ def test_column_letter_factory():
         "AC",
         "AD",
     ]
+
+
+@pytest.mark.this
```
Comment (luke-mino-altherr) [inline tests/test_compose/test_write/test_alphabet.py:46]: is this actually needed or leftover?

**Review by luke-mino-altherr — APPROVED** [review]

File: tests/test_compose/test_write/test_alphabet.py (hunk 2 above)
Comment (mitchhung) [reply tests/test_compose/test_write/test_alphabet.py:46]: IDK but it seems fine to keep

**Pull request merged**